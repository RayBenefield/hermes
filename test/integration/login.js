import flame from '@leonardvandriel/flame';
import * as databases from './database'; // eslint-disable-line import/no-unresolved,import/extensions
import uuid from '../utils/mock-uuid';
import random from '../utils/mock-random';
import configureSave from '../utils/save-db';
import configureDomain from '../../src/domain';
import configureFlame from '../../src/lib/local-db';

const db = configureFlame(flame);
const domain = configureDomain({ db, uuid, random });
const save = configureSave(flame); // eslint-disable-line no-unused-vars
beforeEach(() => flame.loadDatabase(undefined));

test('sends back a welcome message and instructions message', () => {
    // Given
    const lead = {
        platform: 'facebook',
        id: 123456,
    };

    // When
    return domain({
        lead,
        action: 'login',
    })
        .then(results => expect({
            results,
            db: flame.get('/'),
        }).toMatchSnapshot());
});

test('welcomes back a player', () => {
    // Given
    const lead = {
        platform: 'facebook',
        id: 123456,
    };
    flame.loadDatabase(databases['existing-player']);

    // When
    return domain({
        lead,
        action: 'login',
    })
        .then(results => expect({
            results,
            db: flame.get('/'),
        }).toMatchSnapshot());
});
