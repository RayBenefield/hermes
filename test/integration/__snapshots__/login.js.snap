// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sends back a welcome message and instructions message 1`] = `
Object {
  "db": Object {
    "players": Object {
      "facebook": Object {
        "123456": Object {
          "id": 123456,
          "platform": "facebook",
        },
      },
    },
  },
  "results": Object {
    "action": "login",
    "lead": Object {
      "id": 123456,
      "platform": "facebook",
    },
    "messages": Array [
      Object {
        "payload": Object {
          "id": 123456,
          "platform": "facebook",
        },
        "type": "welcome-message",
      },
      Object {
        "type": "instructions-message",
      },
    ],
    "player": undefined,
  },
}
`;
